webpack/assets/javascripts/react_app/components/notifications/NotificationDrawerToggle.js

```diff
@@ -23,10 +24,18 @@ class NotificationDrawerToggle extends Component {
   onChange(actionType) {
     switch (actionType) {
       case ACTIONS.RECEIVED_NOTIFICATIONS: {
-        this.setState({
-          count: NotificationsStore.getNotifications().length,
-          isLoaded: true
-        });
+        const notifications = NotificationsStore.getNotifications();
+
+        const newState = _.reduce(notifications, (curr, next) => {
```

Comment:
maybe i'm missing something, but why do we care about the current count ? just set the new state regardless ?